lib/galaxy/web/framework/webapp.py

```diff
@@ -961,17 +962,30 @@ def build_native_uwsgi_app(paste_factory, config_section):
     """uwsgi can load paste factories with --ini-paste, but this builds non-paste uwsgi apps.
 
     In particular these are useful with --yaml or --json for config."""
+    '''
     import uwsgi
     uwsgi_opt = uwsgi.opt
     config_file = uwsgi_opt.get("yaml") or uwsgi_opt.get("json")
+    # legacy, support loading ini uWSGI config without --ini-paste but with the app config under Paste's [app:main] section
+    if config_file is None and uwsgi_opt.get("ini"):
+        config_file = uwsgi_opt.get("ini")
+        parser = nice_config_parser(config_file)
+        if not parser.has_section(config_section) and parser.has_section("app:main"):
+            config_section = "app:main"
+    # support no uWSGI config file or separate app config file, requires setting galaxy_config_file in the uWSGI config
+    config_file = uwsgi_opt.get("galaxy_config_file") or config_file
+
     if not config_file:
         # Probably loaded via --ini-paste - expect paste app.
         return None
+    '''
+    app_kwds = get_app_kwds(config_section, for_paste_app=True)
+    if not app_kwds:
+        # Probably loaded via --ini-paste - expect paste app
+        return None
```

Comment:
This seems wrong - uwsgi is dying for me with a ``NoneType is not callable`` error not attached to any particular line or stack trace because of this line. Probably because I'm not setting a config file - I don't think that should really be needed but if it is I think we should die with a clearer error here.

Reply:
This was how it was before, but yeah, in expanding this I had a lot of difficulty getting this right. I'll take another look.

It's possible to run under uWSGI with no config file. Can you send the args you were using to get the error you got?

Reply:
This was before I was serializing the YAML in the test case - so it was the test case without the ``--yaml`` argument.

Reply:
Yeah, I found it. I'd had it working but broke it before I committed and didn't realize it.